runtime/src/main/cpp/Memory.cpp

```diff
@@ -1188,28 +1188,27 @@ OBJ_GETTER(InitSharedInstance,
     // OK'ish, inited by someone else.
     RETURN_OBJ(value);
   }
-
   ObjHeader* object = AllocInstance(type_info, OBJ_RESULT);
-  MEMORY_LOG("Calling UpdateRef from InitInstance\n")
+  RuntimeAssert(object->container()->normal() , "Shared object cannot be co-allocated");
+  MEMORY_LOG("Calling UpdateRef from InitSharedInstance\n")
   UpdateRef(localLocation, object);
 #if KONAN_NO_EXCEPTIONS
   ctor(object);
-  if (!object->container()->frozen())
-    ThrowFreezingException();
+  FreezeSubgraph(object);
```

Comment:
So it makes sense to remove explicit freezing from enums, isn't it?

Reply:
Yep.